### PR TITLE
Wodes grimoire: Initial Release

### PR DESCRIPTION
**Pull request by BloodSpartan**
Adds 10 new spells + their icons

I think I did this right and I'm glad to finally be able to publish this!

### CODE REVIEW

**Comment by BloodSpartan** [comment]
Most modifier spells have addModifier on both client and server, so they could be merged. Regenerate needed to have its variables set into the modifier instead of being in the global module.

File: Wodes_Grimoire/cards/Ensnare.ts [hunk 1]
```diff
@@ -25,57 +23,33 @@ const spell: Spell = {
...
+                for (let unit of targets) {
+                    Unit.addModifier(unit, card.id, underworld, prediction, quantity);
+                }
             }
             if (!prediction && !globalThis.headless) {
                 await new Promise((resolve) => {
-                    setTimeout(resolve, 100); //Very unfamiliar with promises and async functions. 
+                    setTimeout(resolve, 100);
                 })
             }
```
Comment (jordan-octavia-blue) [inline Wodes_Grimoire/cards/Ensnare.ts]: This whole block is unnecessary.  Promises and async are just ways of making something happen "later" rather than one line after another during code execution.

This block essentially says, "wait 100 milliseconds then return the function".  This was only necessary for Undead Blade and Slash to match the animation.

```
                            setTimeout(() => {
                                Unit.takeDamage(unit, damageDone, state.casterUnit, underworld, prediction, state);
                            }, 100)
```

In retrospect, Undead Blade is a little more complicated than most spells and I shouldn't have made it the first mod example.

TL;DR you can remove this block, it just makes the function wait 100 ms before returning.

File: Wodes_Grimoire/cards/FlameStrike.ts
Comment (jordan-octavia-blue) [inline Wodes_Grimoire/cards/FlameStrike.ts:1]: Well done! As I see you did here, the prediction resolved immediately while the non-prediction resolves when the particles are done.  Fantastic!

File: Wodes_Grimoire/cards/Pacify.ts
Comment (jordan-octavia-blue) [inline Wodes_Grimoire/cards/Pacify.ts:1]: Maybe Pacify could set `mana` and `maxMana` of casters to 0 so they can't cast too? Or is it only supposed to effect melee units?

Comment (BloodSpartan) [reply Wodes_Grimoire/cards/Pacify.ts:1]: I have it to specifically not target the support class of units, like priest and summoner because their actions on their turn aren't technically "attacks", it should still work against ranged units like archer and glop. Setting mana/max to 0 as well would also mess with mana steal and burns.

Comment (jordan-octavia-blue) [reply Wodes_Grimoire/cards/Pacify.ts:1]: Ah yes, that makes sense!
